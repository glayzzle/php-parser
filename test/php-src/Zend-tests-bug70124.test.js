// eslint-disable prettier/prettier
const parser = require("../main");

describe("php-src tests", function () {
  // Zend/tests/bug70124.phpt
  it("Bug #70124 (null ptr deref / seg fault in ZEND_HANDLE_EXCEPTION_SPEC_HANDLER)", function () {
    expect(parser.parseCode("<?php\ntry  {\n    echo base_convert([array_search(chr(48),chr(48),chr(48),chr(48),chr(48),$f(\"test\"))],chr(48));\n} catch (Error $e) {\n    var_dump($e->getMessage());\n}\nclass A {\n}\ntry  {\n    echo base_convert([array_search(chr(48),chr(48),chr(48),chr(48),chr(48),a::y(\"test\"))],chr(48));\n} catch (Error $e) {\n    var_dump($e->getMessage());\n}\n$a = new A;\ntry  {\n    echo base_convert([array_search(chr(48),chr(48),chr(48),chr(48),chr(48),$a->y(\"test\"))],chr(48));\n} catch (Error $e) {\n    var_dump($e->getMessage());\n}\ntry  {\n    echo base_convert([array_search(chr(48),chr(48),chr(48),chr(48),chr(48),\\bar\\y(\"test\"))],chr(48));\n} catch (Error $e) {\n    var_dump($e->getMessage());\n}\ntry  {\n    echo base_convert([array_search(chr(48),chr(48),chr(48),chr(48),chr(48),y(\"test\"))],chr(48));\n} catch (Error $e) {\n    var_dump($e->getMessage());\n}\n?>")).toMatchSnapshot();
  });
});
