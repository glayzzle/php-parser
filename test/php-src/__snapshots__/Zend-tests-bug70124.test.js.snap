// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`php-src tests Bug #70124 (null ptr deref / seg fault in ZEND_HANDLE_EXCEPTION_SPEC_HANDLER) 1`] = `
Program {
  "children": Array [
    Try {
      "always": null,
      "body": Block {
        "children": Array [
          Echo {
            "expressions": Array [
              Call {
                "arguments": Array [
                  Array {
                    "items": Array [
                      Entry {
                        "byRef": false,
                        "key": null,
                        "kind": "entry",
                        "unpack": false,
                        "value": Call {
                          "arguments": Array [
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                String {
                                  "isDoubleQuote": true,
                                  "kind": "string",
                                  "raw": "\\"test\\"",
                                  "unicode": false,
                                  "value": "test",
                                },
                              ],
                              "kind": "call",
                              "what": Variable {
                                "curly": false,
                                "kind": "variable",
                                "name": "f",
                              },
                            },
                          ],
                          "kind": "call",
                          "what": Name {
                            "kind": "name",
                            "name": "array_search",
                            "resolution": "uqn",
                          },
                        },
                      },
                    ],
                    "kind": "array",
                    "shortForm": true,
                  },
                  Call {
                    "arguments": Array [
                      Number {
                        "kind": "number",
                        "value": "48",
                      },
                    ],
                    "kind": "call",
                    "what": Name {
                      "kind": "name",
                      "name": "chr",
                      "resolution": "uqn",
                    },
                  },
                ],
                "kind": "call",
                "what": Name {
                  "kind": "name",
                  "name": "base_convert",
                  "resolution": "uqn",
                },
              },
            ],
            "kind": "echo",
            "shortForm": false,
          },
        ],
        "kind": "block",
      },
      "catches": Array [
        Catch {
          "body": Block {
            "children": Array [
              ExpressionStatement {
                "expression": Call {
                  "arguments": Array [
                    Call {
                      "arguments": Array [],
                      "kind": "call",
                      "what": PropertyLookup {
                        "kind": "propertylookup",
                        "offset": Identifier {
                          "kind": "identifier",
                          "name": "getMessage",
                        },
                        "what": Variable {
                          "curly": false,
                          "kind": "variable",
                          "name": "e",
                        },
                      },
                    },
                  ],
                  "kind": "call",
                  "what": Name {
                    "kind": "name",
                    "name": "var_dump",
                    "resolution": "uqn",
                  },
                },
                "kind": "expressionstatement",
              },
            ],
            "kind": "block",
          },
          "kind": "catch",
          "variable": Variable {
            "curly": false,
            "kind": "variable",
            "name": "e",
          },
          "what": Array [
            Name {
              "kind": "name",
              "name": "Error",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "kind": "try",
    },
    Class {
      "attrGroups": Array [],
      "body": Array [],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "A",
      },
    },
    Try {
      "always": null,
      "body": Block {
        "children": Array [
          Echo {
            "expressions": Array [
              Call {
                "arguments": Array [
                  Array {
                    "items": Array [
                      Entry {
                        "byRef": false,
                        "key": null,
                        "kind": "entry",
                        "unpack": false,
                        "value": Call {
                          "arguments": Array [
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                String {
                                  "isDoubleQuote": true,
                                  "kind": "string",
                                  "raw": "\\"test\\"",
                                  "unicode": false,
                                  "value": "test",
                                },
                              ],
                              "kind": "call",
                              "what": StaticLookup {
                                "kind": "staticlookup",
                                "offset": Identifier {
                                  "kind": "identifier",
                                  "name": "y",
                                },
                                "what": Name {
                                  "kind": "name",
                                  "name": "a",
                                  "resolution": "uqn",
                                },
                              },
                            },
                          ],
                          "kind": "call",
                          "what": Name {
                            "kind": "name",
                            "name": "array_search",
                            "resolution": "uqn",
                          },
                        },
                      },
                    ],
                    "kind": "array",
                    "shortForm": true,
                  },
                  Call {
                    "arguments": Array [
                      Number {
                        "kind": "number",
                        "value": "48",
                      },
                    ],
                    "kind": "call",
                    "what": Name {
                      "kind": "name",
                      "name": "chr",
                      "resolution": "uqn",
                    },
                  },
                ],
                "kind": "call",
                "what": Name {
                  "kind": "name",
                  "name": "base_convert",
                  "resolution": "uqn",
                },
              },
            ],
            "kind": "echo",
            "shortForm": false,
          },
        ],
        "kind": "block",
      },
      "catches": Array [
        Catch {
          "body": Block {
            "children": Array [
              ExpressionStatement {
                "expression": Call {
                  "arguments": Array [
                    Call {
                      "arguments": Array [],
                      "kind": "call",
                      "what": PropertyLookup {
                        "kind": "propertylookup",
                        "offset": Identifier {
                          "kind": "identifier",
                          "name": "getMessage",
                        },
                        "what": Variable {
                          "curly": false,
                          "kind": "variable",
                          "name": "e",
                        },
                      },
                    },
                  ],
                  "kind": "call",
                  "what": Name {
                    "kind": "name",
                    "name": "var_dump",
                    "resolution": "uqn",
                  },
                },
                "kind": "expressionstatement",
              },
            ],
            "kind": "block",
          },
          "kind": "catch",
          "variable": Variable {
            "curly": false,
            "kind": "variable",
            "name": "e",
          },
          "what": Array [
            Name {
              "kind": "name",
              "name": "Error",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "kind": "try",
    },
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "curly": false,
          "kind": "variable",
          "name": "a",
        },
        "operator": "=",
        "right": New {
          "arguments": Array [],
          "kind": "new",
          "what": Name {
            "kind": "name",
            "name": "A",
            "resolution": "uqn",
          },
        },
      },
      "kind": "expressionstatement",
    },
    Try {
      "always": null,
      "body": Block {
        "children": Array [
          Echo {
            "expressions": Array [
              Call {
                "arguments": Array [
                  Array {
                    "items": Array [
                      Entry {
                        "byRef": false,
                        "key": null,
                        "kind": "entry",
                        "unpack": false,
                        "value": Call {
                          "arguments": Array [
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                String {
                                  "isDoubleQuote": true,
                                  "kind": "string",
                                  "raw": "\\"test\\"",
                                  "unicode": false,
                                  "value": "test",
                                },
                              ],
                              "kind": "call",
                              "what": PropertyLookup {
                                "kind": "propertylookup",
                                "offset": Identifier {
                                  "kind": "identifier",
                                  "name": "y",
                                },
                                "what": Variable {
                                  "curly": false,
                                  "kind": "variable",
                                  "name": "a",
                                },
                              },
                            },
                          ],
                          "kind": "call",
                          "what": Name {
                            "kind": "name",
                            "name": "array_search",
                            "resolution": "uqn",
                          },
                        },
                      },
                    ],
                    "kind": "array",
                    "shortForm": true,
                  },
                  Call {
                    "arguments": Array [
                      Number {
                        "kind": "number",
                        "value": "48",
                      },
                    ],
                    "kind": "call",
                    "what": Name {
                      "kind": "name",
                      "name": "chr",
                      "resolution": "uqn",
                    },
                  },
                ],
                "kind": "call",
                "what": Name {
                  "kind": "name",
                  "name": "base_convert",
                  "resolution": "uqn",
                },
              },
            ],
            "kind": "echo",
            "shortForm": false,
          },
        ],
        "kind": "block",
      },
      "catches": Array [
        Catch {
          "body": Block {
            "children": Array [
              ExpressionStatement {
                "expression": Call {
                  "arguments": Array [
                    Call {
                      "arguments": Array [],
                      "kind": "call",
                      "what": PropertyLookup {
                        "kind": "propertylookup",
                        "offset": Identifier {
                          "kind": "identifier",
                          "name": "getMessage",
                        },
                        "what": Variable {
                          "curly": false,
                          "kind": "variable",
                          "name": "e",
                        },
                      },
                    },
                  ],
                  "kind": "call",
                  "what": Name {
                    "kind": "name",
                    "name": "var_dump",
                    "resolution": "uqn",
                  },
                },
                "kind": "expressionstatement",
              },
            ],
            "kind": "block",
          },
          "kind": "catch",
          "variable": Variable {
            "curly": false,
            "kind": "variable",
            "name": "e",
          },
          "what": Array [
            Name {
              "kind": "name",
              "name": "Error",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "kind": "try",
    },
    Try {
      "always": null,
      "body": Block {
        "children": Array [
          Echo {
            "expressions": Array [
              Call {
                "arguments": Array [
                  Array {
                    "items": Array [
                      Entry {
                        "byRef": false,
                        "key": null,
                        "kind": "entry",
                        "unpack": false,
                        "value": Call {
                          "arguments": Array [
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                String {
                                  "isDoubleQuote": true,
                                  "kind": "string",
                                  "raw": "\\"test\\"",
                                  "unicode": false,
                                  "value": "test",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "\\\\bar\\\\y",
                                "resolution": "fqn",
                              },
                            },
                          ],
                          "kind": "call",
                          "what": Name {
                            "kind": "name",
                            "name": "array_search",
                            "resolution": "uqn",
                          },
                        },
                      },
                    ],
                    "kind": "array",
                    "shortForm": true,
                  },
                  Call {
                    "arguments": Array [
                      Number {
                        "kind": "number",
                        "value": "48",
                      },
                    ],
                    "kind": "call",
                    "what": Name {
                      "kind": "name",
                      "name": "chr",
                      "resolution": "uqn",
                    },
                  },
                ],
                "kind": "call",
                "what": Name {
                  "kind": "name",
                  "name": "base_convert",
                  "resolution": "uqn",
                },
              },
            ],
            "kind": "echo",
            "shortForm": false,
          },
        ],
        "kind": "block",
      },
      "catches": Array [
        Catch {
          "body": Block {
            "children": Array [
              ExpressionStatement {
                "expression": Call {
                  "arguments": Array [
                    Call {
                      "arguments": Array [],
                      "kind": "call",
                      "what": PropertyLookup {
                        "kind": "propertylookup",
                        "offset": Identifier {
                          "kind": "identifier",
                          "name": "getMessage",
                        },
                        "what": Variable {
                          "curly": false,
                          "kind": "variable",
                          "name": "e",
                        },
                      },
                    },
                  ],
                  "kind": "call",
                  "what": Name {
                    "kind": "name",
                    "name": "var_dump",
                    "resolution": "uqn",
                  },
                },
                "kind": "expressionstatement",
              },
            ],
            "kind": "block",
          },
          "kind": "catch",
          "variable": Variable {
            "curly": false,
            "kind": "variable",
            "name": "e",
          },
          "what": Array [
            Name {
              "kind": "name",
              "name": "Error",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "kind": "try",
    },
    Try {
      "always": null,
      "body": Block {
        "children": Array [
          Echo {
            "expressions": Array [
              Call {
                "arguments": Array [
                  Array {
                    "items": Array [
                      Entry {
                        "byRef": false,
                        "key": null,
                        "kind": "entry",
                        "unpack": false,
                        "value": Call {
                          "arguments": Array [
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                Number {
                                  "kind": "number",
                                  "value": "48",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "chr",
                                "resolution": "uqn",
                              },
                            },
                            Call {
                              "arguments": Array [
                                String {
                                  "isDoubleQuote": true,
                                  "kind": "string",
                                  "raw": "\\"test\\"",
                                  "unicode": false,
                                  "value": "test",
                                },
                              ],
                              "kind": "call",
                              "what": Name {
                                "kind": "name",
                                "name": "y",
                                "resolution": "uqn",
                              },
                            },
                          ],
                          "kind": "call",
                          "what": Name {
                            "kind": "name",
                            "name": "array_search",
                            "resolution": "uqn",
                          },
                        },
                      },
                    ],
                    "kind": "array",
                    "shortForm": true,
                  },
                  Call {
                    "arguments": Array [
                      Number {
                        "kind": "number",
                        "value": "48",
                      },
                    ],
                    "kind": "call",
                    "what": Name {
                      "kind": "name",
                      "name": "chr",
                      "resolution": "uqn",
                    },
                  },
                ],
                "kind": "call",
                "what": Name {
                  "kind": "name",
                  "name": "base_convert",
                  "resolution": "uqn",
                },
              },
            ],
            "kind": "echo",
            "shortForm": false,
          },
        ],
        "kind": "block",
      },
      "catches": Array [
        Catch {
          "body": Block {
            "children": Array [
              ExpressionStatement {
                "expression": Call {
                  "arguments": Array [
                    Call {
                      "arguments": Array [],
                      "kind": "call",
                      "what": PropertyLookup {
                        "kind": "propertylookup",
                        "offset": Identifier {
                          "kind": "identifier",
                          "name": "getMessage",
                        },
                        "what": Variable {
                          "curly": false,
                          "kind": "variable",
                          "name": "e",
                        },
                      },
                    },
                  ],
                  "kind": "call",
                  "what": Name {
                    "kind": "name",
                    "name": "var_dump",
                    "resolution": "uqn",
                  },
                },
                "kind": "expressionstatement",
              },
            ],
            "kind": "block",
          },
          "kind": "catch",
          "variable": Variable {
            "curly": false,
            "kind": "variable",
            "name": "e",
          },
          "what": Array [
            Name {
              "kind": "name",
              "name": "Error",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "kind": "try",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
